src/WaterPokemon.java

```diff
@@ -0,0 +1,32 @@
+import java.util.ArrayList;
+
+public class WaterPokemon extends Pokemon {
+
+    public static ArrayList<WaterPokemon> waterPokemons = new ArrayList<>();
```

Comment:
Ik moest even nadenken over waarom en hoe je deze lijsten hebt ingevoegd, maar ze zijn static en daarom zijn ze nuttig. Omdat je in je constructor meteen de pokemon aan deze lijst toevoegd, hoef je ze in je main niet op te slaan in een variabele. Het nadeel daarvan is wel dat je de pokemons niet makkelijk individueel kan aanspreken. Als je 100 Firepokemon hebt, hoe ga je dan een specifieke FirePokemon aanspreken?

Ik vind dit wel een heel ingenieuze oplossing. 👍 

Mischien wil je deze lijsten wel private maken, dat je ze aan kunt passen en ophalen via methodes waarin je eventueel nog checks kunt inbouwen. Anders kunnen gebruikers direct de lijst aanspreken en aanpassen en dat is een potentieel risico.